src/browser-lib/user-fetcher/user-fetcher.ts

```diff
@@ -0,0 +1,38 @@
+import { UserProfile } from "@auth0/nextjs-auth0/client";
+import posthog from "posthog-js";
+
+/**
+ * Fetches the user profile if the feature flag `use-auth-owa` is enabled
+ *
+ * this is a sacrificial module that will be removed once the `use-auth-owa` feature flag is removed
+ * the intent is to prevent the user profile from being fetched by `UserProvider` if the feature flag is not enabled
+ */
+export function featureFlaggedUserFetcher(
```

Comment:
This is a temporary measure. The `UserProvider` is installed in `_app` and by default it will make a request to `/api/auth/me` to see if there is a user logged in. We don't really want this overhead in production until we can measure the impact — so this can be dropped in to replace the default "fetcher" and will only make a request if the feature is enable in Posthog.